### ISSUE TITLE
Amazon: Disable SourceDestCheck on creating AMI

### ISSUE THREAD

**Issue by alaa** [hashicorp/packer#5239]
Because of the increased usage of SDN (Software Defined Networking) solutions for the Container Orchestration systems. Some solutions require all agents to set the network option `SourceDestCheck=false` so it allows the agents to act as `NAT` instances.

However as this option is usually set using Terraform upon launching the (NAT) instance. I have a use case that requires Spot-Instances to have `SourceDestCheck=false` on boot so the agents can join the SDN solution I am using -- in this case: `Calico` --.

I checked Packer documentation and did not find a way to set that option. Please Advice. Thank you.

**Comment by mwhooker** [comment]
hmm, not sure I follow. do you need the source instance to have `source_dest_check` set? i.e. the instance AMI is create from.

**Comment by alaa** [comment]
Sorry if my question wasn't clear enough. Yes exactly I need the source instance to have `source_dest_check` set.

**Comment by mwhooker** [comment]
gotcha. should be fairly easy if anyone wants to take a stab at it I think.

Probably should be added to `StepModifyEBSBackedInstance`

**Comment by alaa** [comment]
I tried to work a bit on this. It seems that `RegisterImageInput struct` on `service/ec2/api.go#L43239` does not have `SourceDestCheck` field But it has for example `EnaSupport`. Which made me wonder if this is even possible by AWS APIs. So I created an EC2 instance, disabled the `SourceDestCheck` manually and created AMI out of it. and launched new one form that AMI. The new instance did not have that attribute `SourceDestCheck` disabled.
